micromanager_gui/_core_widgets/_presets_widget.py

```diff
@@ -0,0 +1,128 @@
+            raise ValueError(f"{self._group} group does not exist.")
+
+        self._presets = list(self._mmc.getAvailableConfigs(self._group))
+
+        if not self._presets:
+            raise ValueError(f"{self._group} group does not have presets.")
+
+        self._combo = QComboBox()
+        self._combo.addItems(self._presets)
+
+        self.setLayout(QHBoxLayout())
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        self.layout().addWidget(self._combo)
+        _set_wdg_color(self.text_color, self._combo)
+        self._combo.currentTextChanged.connect(self._on_combo_changed)
+        self._combo.textActivated.connect(self._on_text_activate)
+
+        self._mmc.events.configSet.connect(self._on_cfg_set)
+        self._mmc.events.configSet.connect(self._highlight_if_preset_state_changed)
+        self._mmc.events.systemConfigurationLoaded.connect(self.refresh)
+        self._mmc.events.propertyChanged.connect(self._highlight_if_prop_changed)
+
+        self.destroyed.connect(self.disconnect)
+
+    def _on_text_activate(self, text: str):
+        """Used in case there is only one preset"""
+        if len(self._presets) == 1:
+            self._mmc.setConfig(self._group, text)
+            _set_wdg_color(self.text_color, self._combo)
+
+    def _on_combo_changed(self, text: str) -> None:
+        self._mmc.setConfig(self._group, text)
+        _set_wdg_color(self.text_color, self._combo)
+
+    def _on_cfg_set(self, group: str, preset: str) -> None:
+        if group == self._group and self._combo.currentText() != preset:
+            with signals_blocked(self._combo):
+                self._combo.setCurrentText(preset)
+                _set_wdg_color(self.text_color, self._combo)
+
+    def value(self) -> str:
+        return self._combo.currentText()
+
+    def setValue(self, value: str) -> None:
+        if value not in self._mmc.getAvailableConfigs(self._group):
+            raise ValueError(
+                f"{value!r} must be one of {self._mmc.getAvailableConfigs(self._group)}"
+            )
+        self._combo.setCurrentText(str(value))
+
+    def allowedValues(self) -> Tuple[str]:
+        return tuple(self._combo.itemText(i) for i in range(self._combo.count()))
+
+    def refresh(self) -> None:
+        with signals_blocked(self._combo):
+            self._combo.clear()
+            if self._group not in self._mmc.getAvailableConfigGroups():
+                self._combo.addItem(f"No group named {self._group}.")
+                self._combo.setEnabled(False)
+            else:
+                presets = self._mmc.getAvailableConfigs(self._group)
+                self._combo.addItems(presets)
+                self._combo.setEnabled(True)
+
+    def _highlight_if_preset_state_changed(self, group: str, preset: str):
+        """Set the text color to magenta if a preset has changed"""
+        if group != self._group and not self._mmc.getCurrentConfig(self._group):
+            _set_wdg_color("magenta", self._combo)
```

Comment:
What's meant to be happening here? I'm having trouble following, it seems like we set the color if some other config group has it's preset set?

Reply:
@ianhi  Yes, that's what I wanted to do:

group A -> `preset a `-> (dev1 prop1 `val1a`), (dev2 prop2 val2a), ...
group B -> `preset b `-> (dev1 prop1 `val1b`), ...

if `preset b `is set, and `val1b != val1a`, then `preset a` will be colored in magenta

Reply:
ah - I thnk that the core `configGroupChanged` already does that logic for you :)